demos/object_detection_demo/jupyter_python/object_detection_demo.ipynb

```diff
@@ -0,0 +1,754 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "vocational-blade",
+   "metadata": {},
+   "source": [
+    "## Imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "terminal-internship",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import glob\n",
+    "import json\n",
+    "import os.path\n",
+    "import random\n",
+    "import re\n",
+    "import subprocess\n",
+    "import sys\n",
+    "from pathlib import Path\n",
+    "from time import perf_counter\n",
+    "\n",
+    "import cv2\n",
+    "import ipywidgets as widgets\n",
+    "import matplotlib.pyplot as plt\n",
+    "from IPython.display import HTML, clear_output\n",
+    "from ipywidgets import Layout, interact_manual\n",
+    "from openvino.inference_engine import IECore\n",
+    "\n",
+    "from detection_utils import ColorPalette, download_video, get_model, put_highlighted_text\n",
+    "\n",
+    "# Add the Open Model Zoo common python folder to the path, to import the pipelines module.\n",
+    "open_model_zoo_path = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(os.curdir))))\n",
+    "sys.path.append(os.path.join(open_model_zoo_path, \"demos\", \"common\", \"python\"))\n",
+    "\n",
+    "from pipelines import AsyncPipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "connected-implementation",
+   "metadata": {},
+   "source": [
+    "## Settings\n",
+    "\n",
+    "Set the file and directory paths. The default settings expect that the models are located in `open_model_zoo_models` in your `$HOME` directory, typically `C:\\Users\\username` or `/home/username`. You can change this by setting the `base_model_dir` variable to another directory. Set `models_file` to `models-all.lst` to use all supported models, instead of a subset. This wil increase model download and conversion time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "superb-nancy",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## File settings\n",
+    "# Directory that contains the Open Model Zoo models. It has subdirectories \"intel\" and \"public\"\n",
+    "base_model_dir = os.path.expanduser(\"~/open_model_zoo_models\")\n",
+    "# Directory for Open Model Zoo cache files. Caching speeds up subsequent downloads.\n",
+    "omz_cache_dir = os.path.expanduser(\"~/open_model_zoo_cache\")\n",
+    "models_file = \"models-subset.lst\"  # models-subset.lst contains a subset of supported models.\n",
+    "# models_file = 'models.lst'  # models.lst contains all supported models.\n",
+    "video_directory = \"videos\"\n",
+    "\n",
+    "## Model settings\n",
+    "DEVICE = \"CPU\"\n",
+    "PRECISION = \"FP16\"\n",
+    "\n",
+    "## Demo settings\n",
+    "DOWNLOAD_MODELS = True  # Use Model Downloader to download models from Open Model Zoo\n",
```

Comment:
Doesn't this mean that the notebook will download the models even when run as part of the test suite? That would be undesirable. Can you make it so that it just uses the models already downloaded by the test runner script?

Reply:
Done.